src/forms/Input/Input.tsx

```diff
@@ -1,17 +1,38 @@
 import * as React from "react";
-
+import { useContext } from "react";
 import { Spacer } from "../../Spacer";
+import { UserTrackerContext } from "../../UserTrackerContext";
 import { InputContainer } from "./styles";
 import { InputProps } from "./types";
 
 export const Input: React.FC<InputProps> = ({
   hasError,
   prefixNode,
   suffixNode,
+  onChange,
+  onFocus,
+  name,
   ...inputElementProps
 }) => {
   const inputRef = React.useRef<HTMLInputElement>(null);
   const classNames = computeInputClasses(hasError, inputElementProps.disabled);
+  const { onEvent } = useContext(UserTrackerContext);
+  const handleChangeEvent = (event: React.FormEvent<HTMLInputElement>) => {
+    setTimeout(() => {
+      onEvent({ type: "formChange", name });
+    });
+    if (onChange) {
+      onChange(event);
+    }
+  };
```

Comment:
I think this would be a great use case to make a custom hook that we could use instead of having to write these out for every function. You could just call this hook once for each event type you want to be able to track.

Could do something like:

```suggestion
  const handleChangeEvent = useEventTracker(onChange, "formChange", name);
```

The `useEventTracker` hook could be something like:
```ts
const useEventTracker = (eventFunction: function | undefined, eventType: UserTrackerEventType, fieldName: string) => {
  const { onEvent } = useContext(UserTrackerContext);
  
  const eventHandler = useCallback((event: React.FormEvent<HTMLInputElement>) => {
    onEvent({ type: eventType, name: fieldName });
    eventFunction?.(event);
  }, [onEvent, eventFunction, eventType, fieldName]);

  return eventHandler;
}
```

Reply:
Neat idea I like this!

Reply:
Makes sense to me!